pkg/osquery/tables/wlan/getbssidps.go

```diff
@@ -0,0 +1,2428 @@
+package wlan
+
+// TODO: move this into a ps1 file for better reading/editing, and load from there.
+const getBSSIDCommandTemplate = `
+function Get-Networks {
+    function Convert-ByteArrayToString {
+        [CmdletBinding()] Param (
+            [Parameter(Mandatory = $True, ValueFromPipeline = $True)] [System.Byte[]] $ByteArray
+            )
+
+        $Encoding  = New-Object System.Text.ASCIIEncoding
+        $Encoding.GetString($ByteArray)
+    }
+
+    Add-Type -Path "{{.NativeCodePath}}"
+    $WlanClient = New-Object NativeWifi.WlanClient
+
+    $WlanClient.Interfaces | ForEach-Object { $_.Scan() }
+
+    #check scan progress for each interface every 50 milliseconds
+    $scanInProgress = "false" 
+    do {
+      $scanInProgress = "false" 
+      $WlanClient.Interfaces | ForEach-Object { 
+        $ip = $_.scanInProgress
+        if ($ip -eq "True") {
+          $scanInProgress = "true" 
+        }
+      }; 
+      Start-Sleep -Milliseconds 50
```

Comment:
50ms seems pretty tight in a loop.

Reply:
Ok, I was unsure on this. Per MS documentation, the wifi interface should complete the scan in 4 seconds or less. When I just had a 2-second sleep in here (instead of polling for completeness) it seemed to work pretty well, but I'm running windows on apple hardware that's probably not indicative of most windows machines out there.